src/Cli/dotnet/commands/dotnet-msbuild/Program.cs

```diff
@@ -1,17 +1,47 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
 using System.CommandLine;
-using System.CommandLine.Parsing;
 using Microsoft.DotNet.Cli;
 
 namespace Microsoft.DotNet.Tools.MSBuild
 {
-    public class MSBuildCommand
+    public class MSBuildCommand : MSBuildForwardingApp
     {
-        public static int Run(string[] args)
+        public MSBuildCommand
+            (IEnumerable<string> msbuildArgs,
+            string msbuildPath = null) 
+             : base(msbuildArgs, msbuildPath)
         {
-            return new MSBuildForwardingApp(args).Execute();
+        }
+
+        public static MSBuildCommand FromArgs(string[] args, string msbuildPath = null)
+        { 
+            var parser = Cli.Parser.Instance;
+            var result = parser.ParseFrom("dotnet msbuild", args);
+            return FromParseResult(result, msbuildPath);
+        }
+
+        public static MSBuildCommand FromParseResult(ParseResult parseResult, string msbuildPath = null)
```

Comment:
@baronfel Just wondering, did you review the changes to this class?  I think these changes make it more like the other command classes we have, so it should be fine, but wouldn't hurt to have you take a look.

Reply:
Not too deeply, no - I've mainly looked at the actual code for emitting the new options section. I can take another pass at the broader set of changes though!